default.nix

```diff
@@ -26,68 +29,70 @@ let
     noTest = pkg: haskell.lib.dontCheck pkg;
 
     hspkgs = haskell.packages.${bootghc};
+    ghc    = haskell.compiler.${bootghc};
 
     ourtexlive =
       nixpkgs.texlive.combine {
         inherit (nixpkgs.texlive) scheme-small collection-xetex fncychap titlesec tabulary varwidth framed capt-of wrapfig needspace dejavu-otf helvetic; };
     fonts = nixpkgs.makeFontsConf { fontDirectories = [ nixpkgs.dejavu_fonts ]; };
     docsPackages = if withDocs then [ python3Packages.sphinx ourtexlive ] else [];
 
-    deps =
+    depsSystem = with stdenv.lib; (
       [ autoconf automake m4
         gmp.dev gmp.out glibcLocales
         ncurses.dev ncurses.out
         perl git file which python3
-        (hspkgs.ghcWithPackages (ps: [ ps.alex ps.happy ]))
         xlibs.lndir  # for source distribution generation
-        cabal-install
         zlib.out
         zlib.dev
       ]
       ++ docsPackages
-      ++ stdenv.lib.optional withLlvm llvm_6
-      ++ stdenv.lib.optional withNuma numactl
-      ++ stdenv.lib.optional withDwarf elfutils
-      ++ stdenv.lib.optional (! stdenv.isDarwin) pxz ;
+      ++ optional  withLlvm            llvm_7
+      ++ optional  withNuma            numactl
+      ++ optional  withDwarf           elfutils
+      ++ optional  (! stdenv.isDarwin) pxz
+      ++ optionals stdenv.isDarwin     [ libiconv
+                                         darwin.libobjc
+                                         darwin.apple_sdk.frameworks.Foundation ]
```

Comment:
Have you tested if this works? When I was adding it in #33 I had to add it to `mkDerivation.buildInputs` specifically, not to `deps`. Not sure why.

Reply:
@int-index, `depsSystem` go to `buildInputs`, indeed -- `depsHaskell` are handled separately.

Reply:
Also, to answer your question directly -- no, I didn't and I'd actually appreciate some Darwin testing..

Reply:
I will give it a go, added to reminders.

Reply:
Thank you!

Reply:
I haven't observed any new testsuite failures with this patch on macOS.

Reply:
Thank you!